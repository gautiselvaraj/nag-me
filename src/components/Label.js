import React from 'react';
import styled from 'styled-components';

const Label = styled.label`
  font-size: ${props => props.small ? '.8rem' : '1rem'};
  line-height: ${props => props.small ? '1.5rem' : '2rem'};
  ${props => props.inline ? 'margin-right: 5px;' : ''};
`;

export default ({children, ...otherProps}) => (
  <Label {...otherProps}>
    {children}
  </Label>
);
