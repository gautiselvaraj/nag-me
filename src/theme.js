export default {
  main:           '#214154',
  mainDarken:     '#083148',
  mainLighten:    '#215a7b',
  black:          '#111111',
  white:          '#f6f6f6',
  greyLightest:   '#efefef',
  greyLighter:    '#dddddd',
  greyLight:      '#a5a5a5',
  grey:           '#898989',
  greyDark:       '#777777',
  greyDarker:     '#595959',
  greyDarkest:    '#222',
}
