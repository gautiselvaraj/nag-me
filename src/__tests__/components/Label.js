import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Label from '../../components/Label';
import theme from '../../theme';
import 'jest-styled-components';

describe('<Label />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Label theme={theme}>Label 1</Label>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have correct element, text, height & font-size based on small prop', () => {
    let label = mount(<Label theme={theme}>Label 1</Label>);
    expect(label.find('label').length).toBe(1);
    expect(label.find('label').text()).toBe('Label 1');
    expect(label.find('label')).toHaveStyleRule('font-size', '1rem');
    expect(label.find('label')).toHaveStyleRule('line-height', '2rem');

    label = mount(
      <Label small theme={theme}>
        Label 1
      </Label>
    );
    expect(label.find('label')).toHaveStyleRule('font-size', '.8rem');
    expect(label.find('label')).toHaveStyleRule('line-height', '1.5rem');
  });
});
