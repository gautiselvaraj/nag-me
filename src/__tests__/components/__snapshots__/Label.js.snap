// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should match snapshot 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 2rem;
}

<label
  className="c0"
>
  Label 1
</label>
`;
